deeposlandia/datagen.py

```diff
@@ -0,0 +1,118 @@
+"""Main method to generate new datasets
+"""
+
+import argparse
+import os
+import sys
+
+import pandas as pd
+
+from deeposlandia import utils
+from deeposlandia.dataset import MapillaryDataset, ShapeDataset
+
+def add_instance_arguments(parser):
+    """Add instance-specific arguments from the command line
+
+    Parameters
+    ----------
+    parser : argparse.ArgumentParser
+        Input parser
+
+    Returns
+    -------
+    argparse.ArgumentParser
+        Modified parser, with additional arguments
+    """
+    parser.add_argument('-a', '--aggregate-label', action='store_true',
+                        help="Aggregate labels with respect to their categories")
+    parser.add_argument('-D', '--dataset',
+                        required=True,
+                        help="Dataset type (either mapillary or shapes)")
+    parser.add_argument('-p', '--datapath',
+                        default="./data",
+                        help="Relative path towards data directory")
+    parser.add_argument('-s', '--image-size',
+                        default=256,
+                        type=int,
+                        help=("Desired size of images (width = height)"))
+    parser.add_argument('-T', '--nb-testing-image',
+                        type=int,
+                        default=5000,
+                        help=("Number of testing images"))
+    parser.add_argument('-t', '--nb-training-image',
+                        type=int,
+                        default=18000,
+                        help=("Number of training images"))
+    parser.add_argument('-v', '--nb-validation-image',
+                        type=int,
+                        default=200,
+                        help=("Number of validation images"))
+    return parser
+
+
+if __name__=='__main__':
+    """Main method: generate a dataset from some instance specifications
```

Comment:
I think the string here is not necessary. This is not a docstring. If we want to keep it, we can turn this content into comments with `#`.

Reply:
The line has been dropped.